### DISCUSSION TITLE
[1.1.444] Marigold depth and new preprocessor structure

### DISCUSSION THREAD

**Discussion by huchenlei** [Mikubill/sd-webui-controlnet#2760] [General]
Marigold depth preprocessor added in 1.1.444. 
https://github.com/huchenlei/sd-webui-controlnet-marigold

We introduce similar preprocessor structure used in SD Forge, which makes it easier for developers to add preprocessor as separate extension instead of making direct change in this repo. Just install the preprocessor like any other A1111 extension.

The reason why we did not add marigold before was because of different versions of diffusers often conflict with each other. Now we make the preprocessor optional, so people can optionally disable the marigold preprocessor extension and continue use controlnet. Original discussion: https://github.com/Mikubill/sd-webui-controlnet/issues/2330

Other user visible change:
Control type radio order will be changed to alphabetical order:
![image](https://github.com/Mikubill/sd-webui-controlnet/assets/20929282/588cbf2f-6f6f-450b-8701-50f914152dc9)

**Comment by stackore0804** [comment]
Hi, thank you for your hard work in advance. Today I updated version 1.1.444. After inserting the image, a connection error occurred. I removed the whole thing and reinstalled it, same thing! How can I solve it? Thank you.
![螢幕擷取畫面 2024-04-18 105621](https://github.com/Mikubill/sd-webui-controlnet/assets/127485035/9bf7c14b-9d95-4cc1-b801-a695fd625183)

**Reply by huchenlei** [reply]
Can you attach full console log?

**Reply by stackore0804** [reply]
Hi, yes, but I seem to have found the reason. There may be two reasons:
1. The parameters I use: --medvram --disable-nan-check --xformers --enable-insecure-extension-access --upcast-sampling --precision full --no-half, I take all parameters Lose
2. The image file should not be png, just jpg.
=================================================================================
venv "E:\stable-diffusion-webui\venv\Scripts\Python.exe"
Python 3.10.6 (tags/v3.10.6:9c7b4bd, Aug  1 2022, 21:53:49) [MSC v.1932 64 bit (AMD64)]
Version: v1.9.0
Commit hash: adadb4e3c7382bf3e4f7519126cd6c70f4f8557b
#######################################################################################################
Initializing Civitai Link
If submitting an issue on github, please provide the below text for debugging purposes:

Python revision: 3.10.6 (tags/v3.10.6:9c7b4bd, Aug  1 2022, 21:53:49) [MSC v.1932 64 bit (AMD64)]
Civitai Link revision: 115cd9c35b0774c90cb9c397ad60ef6a7dac60de
SD-WebUI revision: adadb4e3c7382bf3e4f7519126cd6c70f4f8557b

Checking Civitai Link requirements...

#######################################################################################################
removing old version of tensorrt
loading WD14-tagger reqs from E:\stable-diffusion-webui\extensions\stable-diffusion-webui-wd14-tagger\requirements.txt
Checking WD14-tagger requirements.
Launching Web UI with arguments: --medvram --disable-nan-check --xformers --enable-insecure-extension-access --upcast-sampling --precision full --no-half
==============================================================================
You are running torch 2.0.1+cu118.
The program is tested to work with torch 2.1.2.
To reinstall the desired version, run with commandline flag --reinstall-torch.
Beware that this will cause a lot of large files to be downloaded, as well as
there are reports of issues with training tab on the latest version.

Use --skip-version-check commandline argument to disable this check.
==============================================================================
=================================================================================
You are running xformers 0.0.20.
The program is tested to work with xformers 0.0.23.post1.
To reinstall the desired version, run with commandline flag --reinstall-xformers.

Use --skip-version-check commandline argument to disable this check.
=================================================================================
Civitai Helper: Root Path is: E:\stable-diffusion-webui
Civitai Helper: Get Custom Model Folder
*** Error loading script: sd-TemporalKit-UI.py
    Traceback (most recent call last):
      File "E:\stable-diffusion-webui\modules\scripts.py", line 508, in load_scripts
        script_module = script_loading.load_module(scriptfile.path)
      File "E:\stable-diffusion-webui\modules\script_loading.py", line 14, in load_module
        module_spec.loader.exec_module(module)
      File "<frozen importlib._bootstrap_external>", line 883, in exec_module
      File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
      File "E:\stable-diffusion-webui\extensions\TemporalKit\scripts\sd-TemporalKit-UI.py", line 23, in <module>
        from modules.ui import  create_sampler_and_steps_selection
    ImportError: cannot import name 'create_sampler_and_steps_selection' from 'modules.ui' (E:\stable-diffusion-webui\modules\ui.py)

---
[-] ADetailer initialized. version: 24.4.2, num models: 10
[AddNet] Updating model hashes...
100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 6/6 [00:00<00:00, 6000.43it/s]
[AddNet] Updating model hashes...
100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 6/6 [00:00<?, ?it/s]
ControlNet preprocessor location: E:\stable-diffusion-webui\extensions\sd-webui-controlnet\annotator\downloads
2024-04-18 11:08:31,486 - ControlNet - INFO - ControlNet v1.1.444
2024-04-18 11:08:31,601 - ControlNet - INFO - ControlNet v1.1.444
sd-webui-prompt-all-in-one background API service started successfully.
Civitai: API loaded
== WD14 tagger /gpu:0, uname_result(system='Windows', node='DESKTOP-90JK43G', release='10', version='10.0.22631', machine='AMD64') ==
Loading weights [ac68270450] from E:\stable-diffusion-webui\models\Stable-diffusion\beautifulRealistic_brav5.safetensors
2024-04-18 11:08:32,627 - ControlNet - INFO - ControlNet UI callback registered.
No config file found for ip-adapter-faceid-plus_sd15_lora. You can generate it in the LoRA tab.
Civitai Helper: Settings:
Civitai Helper: max_size_preview: True
Civitai Helper: skip_nsfw_preview: False
Civitai Helper: open_url_with_js: True
Civitai Helper: proxy:
Civitai Helper: use civitai api key: False
[AddNet] Updating model hashes...
100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 15/15 [00:00<00:00, 15001.09it/s]
Creating model from config: E:\stable-diffusion-webui\configs\v1-inference.yaml
Loading VAE weights specified in settings: E:\stable-diffusion-webui\models\VAE\vae-ft-mse-840000-ema-pruned.ckpt
Applying attention optimization: xformers... done.
Model loaded in 2.5s (load weights from disk: 0.7s, create model: 0.2s, apply weights to model: 0.4s, apply float(): 0.5s, load VAE: 0.2s, calculate empty prompt: 0.4s).
[AddNet] Updating model hashes...
100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 15/15 [00:00<00:00, 15008.24it/s]
WARNING:py.warnings:E:\stable-diffusion-webui\extensions\sd-webui-additional-networks\scripts\metadata_editor.py:399: GradioDeprecationWarning: The `style` method is deprecated. Please set these arguments in the constructor instead.
  with gr.Row().style(equal_height=False):

WARNING:py.warnings:E:\stable-diffusion-webui\extensions\sd-webui-additional-networks\scripts\metadata_editor.py:521: GradioDeprecationWarning: The `style` method is deprecated. Please set these arguments in the constructor instead.
  cover_image = gr.Image(

Running on local URL:  http://127.0.0.1:7860

To create a public link, set `share=True` in `launch()`.
Civitai Link: Connecting to Civitai Link Server
Civitai Link: Connected to Civitai Link Server
Civitai: Check resources for missing info files
Civitai Link: Connecting to Civitai Link Server
Civitai: Check resources for missing preview images
Startup time: 27.7s (prepare environment: 7.9s, import torch: 4.6s, import gradio: 0.4s, setup paths: 0.6s, initialize shared: 0.2s, other imports: 0.5s, load scripts: 4.6s, create ui: 2.8s, gradio launch: 2.6s, app_started_callback: 3.5s).
Civitai: Found 6 resources missing preview images
Civitai: Found 6 resources missing info files
Civitai Link: Joined room 503b49500e40e2dda56edd3661fa5998
Civitai Link: Joined room 503b49500e40e2dda56edd3661fa5998
Civitai Link: Presence update: SD: 1, Clients: 0
Civitai Link: Presence update: SD: 0, Clients: 0
Civitai Link: Presence update: SD: 1, Clients: 0
Civitai: No info found on Civitai
Civitai: No preview images found on Civitai
1006
1006
1006
1006

**Reply by huchenlei** [reply]
I don't think the 2 potential reasons you provided cause the issue.

I try to reproduce the issue by setting command line args `--medvram --disable-nan-check --xformers --enable-insecure-extension-access --upcast-sampling --precision full --no-half` and uploading a png image, but there is no error popping up.

Have you tried to run controlnet with all other extensions disabled?

**Reply by stackore0804** [reply]
Okay, I'll give it a try and ask if you have any questions. Thanks.

**Comment by rltgjqmcpgjadyd** [comment]
https://github.com/Mikubill/sd-webui-controlnet/issues/2911
Marigold is still broken

Is this ever going to be fixed?
